### PR TITLE
remove unnecessary loop on selectResponse and add a line to the readme

### PR DESCRIPTION
**Pull request by wadtech**
to explain how to send json request bodies successfully.

### CODE REVIEW

File: lib/canned.js [hunk 1]
```diff
@@ -161,19 +159,18 @@ Canned.prototype.getSelectedResponse = function(responses, content, headers) {
   }
 
   stringifyValues(content);
+  // put the contents of the body and the headers into a big, indexed object.
```
Comment (sideshowcoder) [inline lib/canned.js]: don't need those comments...

File: lib/canned.js [hunk 2]
```diff
@@ -161,19 +159,18 @@ Canned.prototype.getSelectedResponse = function(responses, content, headers) {
...
+  inputObj = cannedUtils.extend({}, content, headers)
 
   responses.forEach(function(response) {
     var metaData = that.parseMetaData(response)
-    var variation = cannedUtils.extend({}, content, headers)
 
-    if(typeof metaData.request !== 'object') return; // nothing to match on
+    if (typeof metaData.request !== 'object') return; // nothing to match on
```
Comment (sideshowcoder) [inline lib/canned.js]: no comment needed.

**Comment by sideshowcoder** [comment]
:+1: in general try to avoid comments as much as necessary, if you feel something is unclear create a function with a good name, comments tend to go out of date way to quickly.

**Comment by wadtech** [comment]
I'm actually usually a fan of more documentation, JS isn't always the easiest thing to read :-)

**Comment by sideshowcoder** [comment]
Sorry I didn't merge this before, it is merged now, I had to rebase so couldn't click the button but it is in master.

**Pull request closed without merging**